GVFS/GVFS.Service/RepoAutoMounter.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.NamedPipes;
+using GVFS.Common.Tracing;
+using GVFS.Service.Handlers;
+
+namespace GVFS.Service
+{
+    public class RepoAutoMounter : IDisposable

```

Comment:
Currently there is no Mac daemon equivalent to the Windows service, but when there is in the future, this class is platform independent.